### ISSUE TITLE
SearchAdapter exception

### ISSUE THREAD

**Issue by alashow** [lapism/SearchView#29]
This exception occasionally:
```
Fatal Exception: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
       at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
       at java.util.ArrayList.get(ArrayList.java:308)
       at com.lapism.searchview.adapter.SearchAdapter.onBindViewHolder(SearchAdapter.java:99)
       at com.lapism.searchview.adapter.SearchAdapter.onBindViewHolder(SearchAdapter.java:25)
       at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:5217)
       at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:5250)
       at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:4487)
       at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:4363)
       at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:1961)
       at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1370)
       at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1333)
       at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:562)
       at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:2900)
       at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:3071)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
       at com.android.internal.policy.PhoneWindow$DecorView.onLayout(PhoneWindow.java:2678)
       at android.view.View.layout(View.java:16636)
       at android.view.ViewGroup.layout(ViewGroup.java:5437)
       at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2171)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1931)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
       at android.view.Choreographer.doCallbacks(Choreographer.java:670)
       at android.view.Choreographer.doFrame(Choreographer.java:606)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5417)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```
http://crashes.to/s/e4f74b7bcff

**Comment by lapism** [comment]
Which version? 1.5?

**Comment by alashow** [comment]
Nope, it's 1.6 (cloned locally until 1.6 release)

But, İ do have suggestions, i'm clearing & re-adding them before calling `#openSearchView`(like in the sample)

**Comment by lapism** [comment]
Ok. 1.6 will not be uploaded cause 2.0 is already out, so check this version and we will see. But still it is not uploaded to jCenter because I use Novoda plugin and it is not working anymore right now. After I will fix method for uploading I will upload it to jCenter. It will be soon, do not worry :).

**Comment by alashow** [comment]
Thanks!

**Comment by alashow** [comment]
Hi there!

Is this bug fixed yet?

I got 4 crashes from Crashlytics: http://crashes.to/s/ad9d0dbd7e3

**Comment by lapism** [comment]
Hi, yes, but it should has been fixed my in my custom version for 3.0 release. Till 3.0 you can try 0.1.4. 
And for 2.2 I am not sure.
